File: create_admin.py
import os
from supabase import create_client, Client
from dotenv import load_dotenv

# Carrega as variáveis de ambiente
load_dotenv('backend/.env')

url: str = os.environ.get("SUPABASE_URL")
key: str = os.environ.get("SUPABASE_SERVICE_ROLE_KEY")

if not url or not key:
    print("ERRO: SUPABASE_URL ou SUPABASE_SERVICE_ROLE_KEY não encontrados no .env do backend")
    exit(1)

supabase: Client = create_client(url, key)

def create_admin(email, password):
    print(f"Tentando criar usuário: {email}...")
    
    # 1. Cria o usuário no Auth (usando admin API do service_role)
    try:
        user = supabase.auth.admin.create_user({
            "email": email,
            "password": password,
            "email_confirm": True
        })
        
        if not user.user:
            print("Erro ao criar usuário no Auth")
            return

        user_id = user.user.id
        print(f"Usuário criado no Auth com ID: {user_id}")

        # 2. Insere na tabela 'users' (caso o trigger não tenha funcionado ou para garantir)
        # Verificando se já existe para evitar erro de PK
        existing = supabase.table("users").select("*").eq("auth_id", user_id).execute()
        
        if len(existing.data) == 0:
            print("Inserindo dados na tabela 'users'...")
            supabase.table("users").insert({
                "auth_id": user_id,
                "email": email,
                "role": "operator",
                "full_name": "Administrador Inicial"
            }).execute()
            print("Perfil criado na tabela 'users'.")
        else:
            print("Perfil já existia na tabela 'users'.")

        print("\nSUCESSO! Você já pode logar com:")
        print(f"E-mail: {email}")
        print(f"Senha: {password}")

    except Exception as e:
        print(f"Erro inesperado: {str(e)}")

if __name__ == "__main__":
    import sys
    email = "admin@facebrasil.com"
    password = "password123"
    
    if len(sys.argv) > 1:
        email = sys.argv[1]
    if len(sys.argv) > 2:
        password = sys.argv[2]
        
    create_admin(email, password)

File: frontend/create_admin.js
const { createClient } = require('@supabase/supabase-js');
const dotenv = require('dotenv');
const path = require('path');

// Carrega as variáveis do .env.local do frontend
dotenv.config({ path: path.join(__dirname, '.env.local') });

const supabaseUrl = process.env.SUPABASE_URL;
const supabaseKey = process.env.SUPABASE_SERVICE_ROLE_KEY || process.env.NEXT_PUBLIC_SUPABASE_ANON_KEY;

// Nota: Para criar usuários confirmados sem e-mail, precisa da SERVICE_ROLE_KEY.
// Se ela não estiver no .env.local, vou pedir para o usuário conferir.

if (!supabaseUrl) {
    console.error("ERRO: SUPABASE_URL não encontrada no .env.local");
    process.exit(1);
}

const serviceRoleKey = process.env.SUPABASE_SERVICE_ROLE_KEY;

if (!serviceRoleKey) {
    console.error("ERRO: SUPABASE_SERVICE_ROLE_KEY não encontrada no .env.local");
    console.log("DICA: Copie a SERVICE_ROLE_KEY do painel do Supabase para o seu .env.local");
    process.exit(1);
}

const supabase = createClient(supabaseUrl, serviceRoleKey, {
    auth: {
        autoRefreshToken: false,
        persistSession: false
    }
});

async function createAdmin(email, password) {
    console.log(`Tentando criar usuário: ${email}...`);

    try {
        // 1. Cria o usuário no Auth
        const { data: authData, error: authError } = await supabase.auth.admin.createUser({
            email: email,
            password: password,
            email_confirm: true
        });

        if (authError) {
            console.error("Erro ao criar usuário no Auth:", authError.message);
            return;
        }

        const userId = authData.user.id;
        console.log(`Usuário criado no Auth com ID: ${userId}`);

        // 2. Insere na tabela 'users'
        const { data: userData, error: userError } = await supabase
            .from('users')
            .upsert({
                auth_id: userId,
                email: email,
                role: 'operator',
                full_name: 'Administrador Inicial'
            })
            .select();

        if (userError) {
            console.error("Erro ao inserir na tabela 'users':", userError.message);
        } else {
            console.log("Perfil criado/atualizado na tabela 'users'.");
        }

        console.log("\nSUCESSO! Você já pode logar com:");
        print(`E-mail: ${email}`);
        print(`Senha: ${password}`);

    } catch (err) {
        console.error("Erro inesperado:", err.message);
    }
}

function print(msg) { console.log(msg); }

const args = process.argv.slice(2);
const email = args[0] || "admin@facebrasil.com";
const password = args[1] || "password123";

createAdmin(email, password);
